pkg/lifecycle/service_test.go

```diff
@@ -250,24 +248,24 @@ func TestServiceLifecycle_PipelineSuccess(t *testing.T) {
 	// create mocked connectors
 	ctrl := gomock.NewController(t)
 	wantRecords := generateRecords(10)
-	source, sourceDispenser := generatorSource(ctrl, persister, wantRecords, nil, true)
-	destination, destDispenser := asserterDestination(ctrl, persister, wantRecords)
-	dlq, dlqDispenser := asserterDestination(ctrl, persister, nil)
+	source, srcDispenser := asserterSource(ctrl, persister, wantRecords, nil, true, 1)
```

Comment:
Is there a special reason for changing to asserterSource?

Reply:
I was wondering if the behavior was different, but looks like not :shrug: . Looks good to me then!